modules/augment/src/rotate.cpp

```diff
@@ -0,0 +1,68 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+namespace cv { namespace augment {
+
+Rotate::Rotate(float _minAngle, float _maxAngle, std::vector<int> _interpolations, std::vector<int> _borderTypes, const Scalar&  _borderValue)
+{
+    minAngle = _minAngle;
+    maxAngle = _maxAngle;
+    angle = minAngle;
+
+    if (_borderTypes.size() > 0)
+        borderTypes = _borderTypes;
+    else
+        borderTypes = { BORDER_CONSTANT , BORDER_REPLICATE , BORDER_WRAP , BORDER_REFLECT_101 };
```

Comment:
What about the rest of border types? https://docs.opencv.org/master/d2/de8/group__core__array.html#ga209f2f4869e304c82d07739337eae7c5

Reply:
The rest of border types when combined with the rest of interpolations they throw exception at warpAffine()

I had to try all the combinations to detect them